core/src/test/java/org/elasticsearch/search/aggregations/bucket/range/InternalBinaryRangeTests.java

```diff
@@ -62,8 +56,17 @@ public void setUp() throws Exception {
         if (randomBoolean()) {
             listOfRanges.add(Tuple.tuple(null, null));
         }
+
+        final int numRanges = randomNumberOfBuckets() - listOfRanges.size();
```

Comment:
Nit: I liked the Math.max() solution you use later better although this also works I think.